[ICU] build failure

### Operating system

Windows 11 Pro

### Compiler

MSVC 17.6.5

### Steps to reproduce the behavior

```Shell
`.\vcpkg.exe install icu:x64-windows`
```
vcpkg is updated to date.


### Failure logs

```powershell
  .\vcpkg.exe install icu:x64-windows
Computing installation plan...
The following packages will be built and installed:
    icu[core,tools]:x64-windows -> 73.1
Detecting compiler hash for triplet x64-windows...
Restored 0 package(s) from E:\_DevEnv\vcpkg_bin_caches in 106 us. Use --debug to see more details.
Installing 1/1 icu:x64-windows...
Building icu[core,tools]:x64-windows...
-- Using cached icu4c-73_1-src.tgz.
-- Cleaning sources at E:/_DevEnv/vcpkg/buildtrees/icu/src/c-73_1-src-3575e21c7c.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source E:/_DevEnv/vcpkg/downloads/icu4c-73_1-src.tgz
-- Applying patch disable-escapestr-tool.patch
-- Applying patch remove-MD-from-configure.patch
-- Applying patch fix_parallel_build_on_windows.patch
-- Applying patch fix-extra.patch
-- Applying patch mingw-dll-install.patch
-- Applying patch disable-static-prefix.patch
-- Applying patch fix-win-build.patch
-- Using source at E:/_DevEnv/vcpkg/buildtrees/icu/src/c-73_1-src-3575e21c7c.clean
-- Getting CMake variables for x64-windows
-- Using cached msys2-autoconf-wrapper-20221207-1-any.pkg.tar.zst.
-- Using cached msys2-automake-wrapper-20221207-1-any.pkg.tar.zst.
-- Using cached msys2-binutils-2.40-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libtool-2.4.7-3-x86_64.pkg.tar.zst.
-- Using cached msys2-make-4.4.1-1-x86_64.pkg.tar.zst.
-- Using cached msys2-pkgconf-1.9.4-1-x86_64.pkg.tar.zst.
-- Using cached msys2-which-2.21-4-x86_64.pkg.tar.zst.
-- Using cached msys2-autoconf-archive-2022.09.03-1-any.pkg.tar.zst.
-- Using cached msys2-bash-5.2.015-1-x86_64.pkg.tar.zst.
-- Using cached msys2-coreutils-8.32-5-x86_64.pkg.tar.zst.
-- Using cached msys2-file-5.44-5-x86_64.pkg.tar.zst.
-- Using cached msys2-gawk-5.2.1-2-x86_64.pkg.tar.zst.
-- Using cached msys2-grep-1~3.0-6-x86_64.pkg.tar.zst.
-- Using cached msys2-gzip-1.12-2-x86_64.pkg.tar.zst.
-- Using cached msys2-diffutils-3.9-1-x86_64.pkg.tar.zst.
-- Using cached msys2-sed-4.9-1-x86_64.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.4.6-1-x86_64.pkg.tar.zst.
-- Using cached msys2-autoconf2.71-2.71-1-any.pkg.tar.zst.
-- Using cached msys2-automake1.16-1.16.5-1-any.pkg.tar.zst.
-- Using cached msys2-libiconv-1.17-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libintl-0.21-2-x86_64.pkg.tar.zst.
-- Using cached msys2-zlib-1.2.13-1-x86_64.pkg.tar.zst.
-- Using cached msys2-tar-1.34-3-x86_64.pkg.tar.zst.
-- Using cached msys2-findutils-4.9.0-3-x86_64.pkg.tar.zst.
-- Using cached msys2-gmp-6.2.1-2-x86_64.pkg.tar.zst.
-- Using cached msys2-gcc-libs-11.3.0-3-x86_64.pkg.tar.zst.
-- Using cached msys2-libbz2-1.0.8-4-x86_64.pkg.tar.zst.
-- Using cached msys2-liblzma-5.4.1-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libzstd-1.5.4-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libreadline-8.2.001-3-x86_64.pkg.tar.zst.
-- Using cached msys2-mpfr-4.2.0-2-x86_64.pkg.tar.zst.
-- Using cached msys2-libpcre-8.45-3-x86_64.pkg.tar.zst.
-- Using cached msys2-m4-1.4.19-2-x86_64.pkg.tar.zst.
-- Using cached msys2-perl-5.36.0-1-x86_64.pkg.tar.zst.
-- Using cached msys2-ncurses-6.4-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libcrypt-2.1-4-x86_64.pkg.tar.zst.
-- Using msys root at E:/_DevEnv/vcpkg/downloads/tools/msys2/87962f0d86055c95
CMake Warning at scripts/cmake/vcpkg_configure_make.cmake:264 (message):
  Unable to find system dir in the PATH variable! Appending required msys
  paths!
Call Stack (most recent call first):
  ports/icu/portfile.cmake:54 (vcpkg_configure_make)
  scripts/ports.cmake:147 (include)


-- Generating configure for x64-windows
-- Finished generating configure for x64-windows
-- Using cached mingw-w64-i686-pkgconf-1~1.8.0-2-any.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.4.6-1-x86_64.pkg.tar.zst.
-- Using msys root at E:/_DevEnv/vcpkg/downloads/tools/msys2/6f3fa1a12ef85a6f
-- Configuring x64-windows-dbg
-- Configuring x64-windows-rel
-- Building x64-windows-dbg
-- Installing x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: E:/_DevEnv/vcpkg/downloads/tools/msys2/87962f0d86055c95/usr/bin/make.exe -j 9 --trace -f Makefile install DESTDIR=/E/_DevEnv/vcpkg/packages/icu_x64-windows
    Working Directory: E:/_DevEnv/vcpkg/buildtrees/icu/x64-windows-dbg/
    See logs for more information:
      E:\_DevEnv\vcpkg\buildtrees\icu\install-x64-windows-dbg-out.log
      E:\_DevEnv\vcpkg\buildtrees\icu\install-x64-windows-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:168 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:2 (vcpkg_build_make)
  ports/icu/portfile.cmake:144 (vcpkg_install_make)
  scripts/ports.cmake:147 (include)


error: building icu:x64-windows failed with: BUILD_FAILED
Elapsed time to handle icu:x64-windows: 4.8 min
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+icu
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?title=[icu]+Build+error&body=Copy+issue+body+from+E%3A%2F_DevEnv%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md

```

**log files**
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/12138854/install-x64-windows-dbg-out.log)
[install-x64-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/12138856/install-x64-windows-dbg-err.log)


### Additional context

_No response_

I moved vcpkg directory to `E:/vcpkg` and re-run `vcpkg.exe install icu:x64-windows`, i got the same error message as below.

```shell
/usr/bin/install: cannot create regular file 'E:/vcpkg/packages/icu_x64-windows/E/vcpkg/installed/x64-windows/debug/lib/icudtd73.dll': No such file or directory
-- return status = 1
Error installing library. Failed command: cd ../lib/ && /usr/bin/install -c icudtd73.dll E:/vcpkg/packages/icu_x64-windows/E/vcpkg/installed/x64-windows/debug/lib/icudtd73.dll
make[1]: *** [Makefile:185: install-local] Error 1
make: *** [Makefile:153: install-recursive] Error 2
```

It seems a wrong path have been made by build system or compiler.

I tried a total new vcpkg environment to reproduce, and solved this error.  But my previous vcpkg has installed many packages, so rebuilding all packages maybe cost many time.

How about clearing possible cache contents to clean up some potential accumulated errors? Which folders can I delete to keep the previous installation, but clean the cache?

Thanks!

--- a/scripts/cmake/vcpkg_configure_make.cmake
+++ b/scripts/cmake/vcpkg_configure_make.cmake
@@ -236,6 +236,7 @@ function(vcpkg_configure_make)
         cmake_path(CONVERT "$ENV{SystemRoot}" TO_CMAKE_PATH_LIST system_root NORMALIZE)
         cmake_path(CONVERT "$ENV{LOCALAPPDATA}" TO_CMAKE_PATH_LIST local_app_data NORMALIZE)
         file(REAL_PATH "${system_root}" system_root)
+        string(TOUPPER "${system_root}" system_root_upper)
 
         message(DEBUG "path_list:${path_list}") # Just to have --trace-expand output
 
@@ -244,6 +245,10 @@ function(vcpkg_configure_make)
             "${system_root}/System32"
             "${system_root}/system32/"
             "${system_root}/System32/"
+            "${system_root_upper}/system32"
+            "${system_root_upper}/System32"
+            "${system_root_upper}/system32/"
+            "${system_root_upper}/System32/"
             "${local_app_data}/Microsoft/WindowsApps"
             "${local_app_data}/Microsoft/WindowsApps/"
         )

